src/NuGetGallery/Filters/ApiAuthorizeAttribute.cs

```
@@ -49,16 +74,47 @@ public override void OnAuthorization(AuthorizationContext filterContext)
                     }
                 }
             }
+            
+            // Resolve the policy service if security policy checks are required.
+            if (SecurityPolicyAction.HasValue)
+            {
+                SecurityPolicyService = ((AppController)filterContext.Controller)?.GetService<ISecurityPolicyService>();
```

This will throw if the `ApiAuthorizeAttribute` is used on a controller that doesn't extend `AppController`, right? I would recommend that you add documentation that this attribute __must__ be used on controllers that extend `AppController`. Also, maybe throw an exception if the controller isn't an `AppController` to make this easier to debug if this issue ever comes up?
*